### PR TITLE
Additional resource types added

### PR DESCRIPTION
**Pull request by rain-on**
The following types have been added to enrich the java sdk:

* LIfecycle
* Channels
* Tasks (& task state)

### CODE REVIEW

File: acceptance-tests/src/test/java/com/octopus/sdk/test/ChannelAcceptanceTest.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import com.octopus.sdk.model.projectgroup.ProjectGroupResource;
+
+import java.io.IOException;
+
+import org.junit.jupiter.api.Test;
+
+public class ChannelAcceptanceTest extends SpaceScopedAcceptanceTest {
+
+  @Test
+  public void canCreateNewChannel() throws IOException {
```
Comment (long-sean) [inline acceptance-tests/src/test/java/com/octopus/sdk/test/ChannelAcceptanceTest.java:33]: Confirm with assertion?

File: octopus-sdk/src/main/java/com/octopus/sdk/domain/Task.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ */
+
+package com.octopus.sdk.domain;
+
+import com.octopus.sdk.http.OctopusClient;
+import com.octopus.sdk.model.task.TaskResourceWithLinks;
+
+public class Task {
+
+  @SuppressWarnings("UnusedVariable")
```
Comment (long-sean) [inline octopus-sdk/src/main/java/com/octopus/sdk/domain/Task.java:23]: Not required?

File: octopus-sdk/src/main/java/com/octopus/sdk/model/lifecycle/LifecycleResource.java [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+import com.octopus.openapi.model.RetentionPeriod;
+import com.octopus.sdk.model.NamedResource;
+
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.Objects;
+
+import com.google.gson.annotations.SerializedName;
+
+/** LifecycleResource */
```
Comment (long-sean) [inline octopus-sdk/src/main/java/com/octopus/sdk/model/lifecycle/LifecycleResource.java:28]: Nit: Comment not needed

File: octopus-sdk/src/main/java/com/octopus/sdk/domain/Task.java (hunk 2 above)
Comment (rain-on) [reply octopus-sdk/src/main/java/com/octopus/sdk/domain/Task.java:23]: true!

File: octopus-sdk/src/main/java/com/octopus/sdk/model/lifecycle/LifecycleResource.java (hunk 3 above)
Comment (rain-on) [reply octopus-sdk/src/main/java/com/octopus/sdk/model/lifecycle/LifecycleResource.java:28]: true

**Review by long-sean — APPROVED** [review]
Good to go

**Pull request merged**